developer/src/kmc-keyboard-info/src/index.ts

```diff
@@ -463,6 +466,39 @@ export class KeyboardInfoCompiler {
         ''
       );
     }
+    return true;
   }
+
+  async fontSourceToKeyboardInfoFont(kpsFilename: string, kmpJsonData: KmpJsonFile.KmpJsonFile, source: string[]): Promise<KeyboardInfoFileLanguageFont> {
+    // locate a .ttf, .otf, or .woff font file
+    const ttf = source.find(file => file.endsWith('.ttf') || file.endsWith('.otf') || file.endsWith('.woff'));
+    if(!ttf) {
+      return {
+        // If we can't find a matching font, we'll just use the filename of the first font
+        family: this.callbacks.path.basename(source[0]),
+        source: source
+      }
+    }
+
+    // The font sources already have path information stripped, but we can
+    // find the matching file from the list of files in the package.
+    const sourcePath = kmpJsonData.files.find(file => ('/' + file.name.replaceAll('\\', '/')).endsWith('/'+ttf))?.name;
+    if(!sourcePath) {
+      this.callbacks.reportMessage(KeyboardInfoCompilerMessages.Error_FileDoesNotExist({filename: ttf}));
+      return null;
+    }
+
+    const result = {
+      family: await getFontFamily(this.callbacks.loadFile(this.callbacks.resolveFilename(kpsFilename, sourcePath))),
+      source
+    };
+
+    if(!result.family) {
+      this.callbacks.reportMessage(KeyboardInfoCompilerMessages.Error_FileDoesNotExist({filename: sourcePath}));
```

Comment:
This needs to be a different error message -- file does exist but can't be parsed